core/classes/Database/DB.php

```diff
@@ -8,29 +8,411 @@
+     */
+    public function orderWhere(string $table, string $where, string $order, string $sort) {
+        $table = $this->_prefix . $table;
+        $sql = "SELECT * FROM {$table} WHERE {$where} ORDER BY {$order} {$sort}";
+
+        if (!$this->query($sql)->error()) {
+            return $this;
+        }
+
+        return false;
+    }
+
+    /**
+     * Create a new table in the database.
+     *
+     * @param string $name The name of the table.
+     * @param string $table_schema The table SQL schema.
+     * @param string $other Other SQL table options.
+     * @return bool Whether an error occurred or not.
+     */
+    public function createTable(string $name, string $table_schema, string $other): bool {
+        $name = $this->_prefix . $name;
+        $sql = "CREATE TABLE `{$name}` ({$table_schema}) {$other}";
+
+        return !$this->query($sql)->error();
+    }
+
+    /**
+     * Perform a SHOW TABLES LIKE query.
+     *
+     * @param string $table Name of table to try and lookup.
+     * @return int|false The number of rows affected, or false on failure.
+     */
+    public function showTables(string $table) {
+        $table = $this->_prefix . $table;
+        $sql = "SHOW TABLES LIKE '{$table}'";
+
+        if (!$this->query($sql)->error()) {
+            return $this->_statement->rowCount();
+        }
+
+        return false;
+    }
+
+    /**
+     * Add a new column to a table.
+     *
+     * @param string $table Name of table to alter.
+     * @param string $column The column to add.
+     * @param string $attributes The attributes of the column.
+     * @return bool Whether an error occurred or not.
+     */
+    public function addColumn(string $table, string $column, string $attributes): bool {
+        $table = $this->_prefix . $table;
+        $sql = "ALTER TABLE {$table} ADD {$column} {$attributes}";
+
+        return !$this->query($sql)->error();
+    }
+
+    /**
+     * Attempt to parse a where clause in array format.
+     *
+     * @internal This method is used by the class and should not be called directly.
+     * @param array $clause The where clause.
+     * @return array The parsed where clause as [field, operator, value] form.
+     */
+    private function parseWhereClause(array $clause): array {
+        $operators = ['=', '>', '<', '>=', '<=', '<>'];
+        if (count($clause) === 3) {
+            [$field, $operator, $value] = $clause;
+        } else if (count($clause) === 2) {
+            [$field, $value] = $clause;
+            $operator = '=';
+        } else {
+            throw new InvalidArgumentException("Invalid SQL where clause");
+        }
+
+        if (!in_array($operator, $operators)) {
+            throw new InvalidArgumentException("Invalid SQL operator: '{$operator}'");
+        }
+
+        return [$field, $operator, $value];
+    }
+
+    public static function getCustomInstance(string $host, string $database, string $username,
```

Comment:
`connection` might be a more fluent word here... what do you think?

Maybe not actually, since then it wouldn't match with `getInstance` 🤔

Reply:
I do like getConnection, but that would be a breaking change (currently, only usage of the custom class changed)